app/services/music_import_service.rb

```diff
@@ -162,7 +162,7 @@ def ingest_fake_playlist
           # Find previously imported dependent file sets
           file_set_ids = selection_files.map do |file|
             {
-              buffered_change_set_persister.query_service.custom_queries.find_by_property(property: :local_identifier, value: file.entry_id.to_s).first&.id => file.id
+              query_service.custom_queries.find_by_property(property: :local_identifier, value: file.entry_id.to_s).first&.id => file.id
```

Comment:
I'm a little surprised this works. I guess Sequel holds the transaction open for a given thread?

Reply:
How would you expect it to break?

Reply:
It's the query service for the non-buffered adapter, which I would have thought would be outside the transaction, but I guess that's not how it works.

Reply:
Okay! Yeah I guess it didn't occur to me to think that it might somehow happen outside the transaction.